tests/tests_geomstats/test_spd_matrices.py

```diff
@@ -19,559 +17,676 @@
+
+        def exp_data(self):
+            smoke_data = [
+                dict(
+                    n=2,
+                    tangent_vec=[[2.0, 0.0], [0.0, 2.0]],
+                    base_point=[[1.0, 0.0], [0.0, 1.0]],
+                    expected=[[4.0, 0.0], [0.0, 4.0]],
+                )
+            ]
+            return self.generate_tests(smoke_data)
+
+        def log_data(self):
+            smoke_data = [
+                dict(
+                    n=2,
+                    point=[[4.0, 0.0], [0.0, 4.0]],
+                    base_point=[[1.0, 0.0], [0.0, 1.0]],
+                    expected=[[2.0, 0.0], [0.0, 2.0]],
+                )
+            ]
+            return self.generate_tests(smoke_data)
+
+        def squared_dist_data(self):
+            smoke_data = [
+                dict(
+                    n=2,
+                    point_a=[[1.0, 0.0], [0.0, 1.0]],
+                    point_b=[[2.0, 0.0], [0.0, 2.0]],
+                    expected=2 + 4 - (2 * 2 * SQRT_2),
+                )
+            ]
+            return self.generate_tests(smoke_data)
+
+        def log_exp_composition_data(self):
+            return self._log_exp_composition_data(SPDMatrices)
+
+        def geodesic_belongs_data(self):
+            return self._geodesic_belongs_data(
+                SPDMatrices,
+            )
+
+        def squared_dist_is_symmetric_data(self):
+            return self._squared_dist_is_symmetric_data(SPDMatrices)
+
+    testing_data = TestDataSPDMetricBuresWasserstein()
+
+    def test_inner_product(self, n, tangent_vec_a, tangent_vec_b, base_point, expected):
+        metric = SPDMetricBuresWasserstein(n)
+        result = metric.inner_product(
+            gs.array(tangent_vec_a), gs.array(tangent_vec_b), gs.array(base_point)
+        )
+        self.assertAllClose(result, gs.array(expected))
+
+    def test_exp(self, n, tangent_vec, base_point, expected):
+        metric = SPDMetricBuresWasserstein(n)
+        result = metric.exp(gs.array(tangent_vec), gs.array(base_point))
+        self.assertAllClose(result, gs.array(expected))
+
+    def test_log(self, n, point, base_point, expected):
+        metric = SPDMetricBuresWasserstein(n)
+        result = metric.log(gs.array(point), gs.array(base_point))
+        self.assertAllClose(result, gs.array(expected))
+
+    def test_squared_dist(self, n, point_a, point_b, expected):
+        metric = SPDMetricBuresWasserstein(n)
+        result = metric.squared_dist(gs.array(point_a), gs.array(point_b))
+        self.assertAllClose(result, gs.array(expected))
+
+    def test_log_exp_composition(self, n, point, base_point):
+        metric = SPDMetricBuresWasserstein(n)
+        log = metric.log(gs.array(point), base_point=gs.array(base_point))
+        result = metric.exp(tangent_vec=log, base_point=gs.array(base_point))
+        self.assertAllClose(result, point, atol=gs.atol * 1000)
+
+    # def test_geodesic_belongs(self, n, initial_point, initial_tangent_vec, t):
```

Comment:
Rm left over comments: I assume this is not working? was it working before?

Reply:
not working anything more than 5